src/modules/commander/PreflightCheck.cpp

```diff
@@ -448,6 +449,47 @@ static bool airspeedCheck(orb_advert_t *mavlink_log_pub, bool optional, bool rep
 	return success;
 }
 
+static bool powerCheck(orb_advert_t *mavlink_log_pub, bool report_fail, bool prearm)
+{
+	bool success = true; // start with a pass and change to a fail if any test fails
+
+	if (prearm) {
+		int system_power_sub = orb_subscribe(ORB_ID(system_power));
+
+		system_power_s system_power = {};
+
+		if (orb_copy(ORB_ID(system_power), system_power_sub, &system_power) == PX4_OK) {
```

Comment:
If there is no system power update, this check will just pass. As I understand it, there will always be one `orb_copy` and therefore we would copy an `avionics_power_rail_voltage` of `0` and fail anyway.

Reply:
I added a time elapsed check as a workaround, which should be fine for now, but still isn't ideal. After this PR I was thinking about centralizing the arming requirements checks. We can make system_power explicitly required or not.